### ISSUE TITLE
Make Deferred a new style class

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5822]
|[<img alt="fijal's avatar" src="https://avatars.githubusercontent.com/u/85944?s=50" width="50" height="50">](https://github.com/fijal)| @fijal reported|
|-|-|
|Trac ID|trac#5822|
|Type|enhancement|
|Created|2012-07-31 19:02:47Z|

PyPy is slightly faster with new style classes. Changing Deferred only (which should not be subclassed) changes twisted_names benchmark from the pypy suite by about 10%. Some others might benefit as well.

<details><summary>Searchable metadata</summary>

```
trac-id__5822 5822
type__enhancement enhancement
reporter__fijal fijal
priority__normal normal
milestone__New_Style_Classes_Only New-Style Classes Only
branch__ 
branch_author__ 
status__closed closed
resolution__wontfix wontfix
component__core core
keywords__ 
time__1343761367000000 1343761367000000
changetime__1577615049135238 1577615049135238
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

Part of [#8243](https://github.com/twisted/twisted/issues/8243)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Would be fixed via feature flag in [#8281](https://github.com/twisted/twisted/issues/8281)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl set status to `closed`|
|-|-|

This is going to end up being the case when we drop Py2 support, so, I'm going to close this.
